src/pages/Glossary/Glossary.tsx

```diff
@@ -0,0 +1,149 @@
+export const getSelectedTagsWrapper = (setSelectedTags: Dispatch<SetStateAction<string[]>>) => {
+  const setSelectedTagsWrapper = (input?: string | string[]) => {
+    if (input === undefined) {
+      setSelectedTags([])
+    } else if (typeof input === 'string') {
+      setSelectedTags(input.split(','))
+    } else {
+      setSelectedTags(input)
+    }
+  }
+
+  return setSelectedTagsWrapper
+}
+
+/**
+ * Glossary presents a page with a list of important terms of the haski project.
+ * It uses the GlossaryList component to present the content and several other components, like a Filter, Searchbar and GlossaryIndex to filter or search for specific entries.
+ * It is also possible to collapse and expand individual entries or all at once to gain further information to every term.
+ * @param props - Props containing the logic to collapse and expand the every glossary entry.
+ * @returns {JSX.Element} - The Glossary page.
+ * @category Pages
+ */
+const Glossary = ({ useGlossary = _useGlossary }: GlossaryProps) => {
+  // Translation
+  const { t } = useTranslation()
+
+  // Deconstructing array of glossary entries into array to prevent testing errors.
+  const glossaryEntries: GlossaryEntryProps[] = [
+    ...(t<string>('pages.glossary.elements', {
+      returnObjects: true
+    }) as GlossaryEntryProps[])
+  ]
+
+  // Deconstructing array of tags into array to prevent testing errors.
+  const tags = [
+    ...(t<string>('pages.glossary.tags', {
+      returnObjects: true
+    }) as string[])
+  ]
+
+  const indexElements = [t('pages.glossary.fundamentals')].concat([
+    'A',
+    'B',
+    'C',
+    'D',
+    'E',
+    'F',
+    'G',
+    'H',
+    'I',
+    'J',
+    'K',
+    'L',
+    'M',
+    'N',
+    'O',
+    'P',
+    'Q',
+    'R',
+    'S',
+    'T',
+    'U',
+    'V',
+    'W',
+    'X',
+    'Y',
+    'Z'
+  ])
+
+  // Logic from hook
+  const { collapseAll, expandAll } = useGlossary()
+
+  // States
+  const [searchQuery, setSearchQuery] = useState('')
+  const [selectedTags, setSelectedTags] = useState<string[]>([])
+  const [selectedIndexElement, setSelectedIndexElement] = useState<string>('')
+  const [expandedList, setExpandedList] = useState<string[]>([])
+
+  return (
+    <Grid container columnSpacing={1} rowSpacing={1}>
+      <Grid item xs={12} sm={12} sx={{ mt: '1rem', mb: '1rem' }}>
+        <Typography variant="h3">{t('pages.glossary.title')}</Typography>
+      </Grid>
+      <Grid item xs={8} sm={6}>
+        <Searchbar label={t('pages.glossary.search')} setSearchQuery={setSearchQuery} timeout={100} />
+      </Grid>
+      <Grid item xs={4} sm={6}>
+        <Filter
+          label={t('pages.glossary.filter')}
+          options={tags}
+          selectedOptions={selectedTags}
+          setSelectedOptions={getSelectedTagsWrapper(setSelectedTags)}
+        />
+      </Grid>
+      <Grid item xs={12} sm={12}>
+        <Box sx={{ display: 'flex', flexWrap: 'wrap', gap: 1 }}>
+          {useMemo(
```

Comment:
Macht es hier vielleicht Sinn statt Use Memo den Index in memo() zu packen? Müsste ja denselben Effekt haben wenn mich nicht alles täuscht.